ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -260,187 +182,182 @@ Tensor convert_torch_tensor_to_tt_tensor(
-                borrowed_buffer::Buffer(data_ptr, num_elements), on_creation_callback, on_destruction_callback);
-            return Tensor(std::move(storage), shape, data_type, Layout::ROW_MAJOR, optional_tile);
-        }
-        /*
-        case DataType::BFLOAT16: {
-            auto data_ptr = reinterpret_cast<bfloat16 *>(np_data_ptr);
-            auto storage = BorrowedStorage(
-                borrowed_buffer::Buffer(data_ptr, num_elements), on_creation_callback, on_destruction_callback);
-            return Tensor(std::move(storage), shape, data_type, Layout::ROW_MAJOR);
-        }
-        */
-        case DataType::BFLOAT8_B: {
-            auto data_ptr = reinterpret_cast<float *>(np_data_ptr);
-            auto data = std::vector<float>(data_ptr, data_ptr + num_elements);
-            auto buffer = owned_buffer::create<float>(std::move(data));
-            auto tile = optional_tile.value_or(Tile());
-            auto tensor = Tensor(OwnedStorage{buffer}, shape, DataType::FLOAT32, Layout::ROW_MAJOR, optional_tile)
-                              .to(Layout::TILE);
-            auto output_float_data = owned_buffer::get_as<float>(tensor).get();
-            auto output_packed_data = pack_fp32_vec_as_bfp8_tiles(
-                output_float_data, /*row_major_input=*/false, /*is_exp_a=*/false, tile);
-            auto output_buffer = owned_buffer::create<uint32_t>(std::move(output_packed_data));
-            return Tensor(std::move(OwnedStorage{std::move(output_buffer)}), shape, data_type, Layout::TILE, tile);
-        }
-        case DataType::BFLOAT4_B: {
-            auto data_ptr = reinterpret_cast<float *>(np_data_ptr);
-            auto data = std::vector<float>(data_ptr, data_ptr + num_elements);
-            auto buffer = owned_buffer::create<float>(std::move(data));
-            auto tile = optional_tile.value_or(Tile());
-            auto tensor = Tensor(OwnedStorage{buffer}, shape, DataType::FLOAT32, Layout::ROW_MAJOR, optional_tile)
-                              .to(Layout::TILE);
-            auto output_float_data = owned_buffer::get_as<float>(tensor).get();
-            auto output_packed_data = pack_fp32_vec_as_bfp4_tiles(
-                output_float_data, /*row_major_input=*/false, /*is_exp_a=*/false, tile);
-            auto output_buffer = owned_buffer::create<uint32_t>(std::move(output_packed_data));
-            return Tensor(std::move(OwnedStorage{std::move(output_buffer)}), shape, data_type, Layout::TILE, tile);
+        num_elements = py::cast<std::size_t>(contiguous_py_tensor.attr("numel")());
+        py_data_ptr = py::cast<std::size_t>(contiguous_py_tensor.attr("data_ptr")());
+    } else if (py::isinstance(py_tensor, np.attr("ndarray"))) {
+        TT_FATAL(enable_borrow, "Owned storage for numpy tensors is untested!");
+
+        contiguous_py_tensor = np.attr("ascontiguousarray")(py_tensor);
+
+        // Override the data type if there is an user-provided one
```

Comment:
This piece can be deduplicated with the functionality above in a separate function.